apps/src/turtle/blocks.js

```diff
@@ -259,35 +259,84 @@ exports.install = function (blockly, blockInstallOptions) {
     }
   };
 
-blockly.Blocks.point_to = {
-    // Block for pointing to a specified direction
-    helpUrl: '',
-    init: function () {
-      this.setHSV(184, 1.00, 0.74);
-      this.appendDummyInput()
-          .appendTitle(msg.pointTo());
-      this.appendDummyInput()
-          .appendTitle(new blockly.FieldTextInput('0', blockly.FieldTextInput.numberValidator), 'DIRECTION')
+  generator.draw_turn_inline = function () {
+    // Generate JavaScript for turning left or right.
+    var value = window.parseFloat(this.getTitleValue('VALUE'));
+    return 'Turtle.' + this.getTitleValue('DIR') +
+        '(' + value + ', \'block_id_' + this.id + '\');\n';
+  };
+
+  function createPointToDirectionBlock(blockName) {
+    return {
+      helpUrl: '',
+      init: function () {
+        this.setHSV(184, 1.00, 0.74);
+        this.appendDummyInput()
+            .appendTitle(msg.pointTo());
+        this.setPreviousStatement(true);
+        this.setInputsInline(true);
+        this.setNextStatement(true);
+        this.setTooltip(msg.pointTo());
+
+        appendToPointToDirectionBlock(blockName, this);
+      }
+    };
+  }
+
+  function appendToPointToDirectionBlock(blockName, block) {
+    if (blockName === 'point_to_direction') {
+      block.appendDummyInput()
+          .appendTitle(new blockly.FieldAngleTextInput('VALUE', '0'),
+              'DIRECTION')
+          .appendTitle(msg.degrees());
+    } else if (blockName === 'point_to_direction_param') {
+      block.appendValueInput('VALUE')
+          .setCheck(blockly.BlockValueType.NUMBER)
+          .addFieldHelper(blockly.BlockFieldHelper.ANGLE_HELPER, {
+            block: block
+          });
+      block.appendDummyInput()
+          .appendTitle(msg.degrees());
+    } else {
+      block.appendDummyInput()
+          .appendTitle(new blockly.FieldAngleDropdown('DIRECTION', block.VALUE),
+              'VALUE')
           .appendTitle(msg.degrees());
-      this.setPreviousStatement(true);
-      this.setInputsInline(true);
-      this.setNextStatement(true);
-      this.setTooltip(msg.pointTo());
     }
-  };
+  }
 
-  generator.point_to = function () {
+  blockly.Blocks.point_to_direction =
+      createPointToDirectionBlock('point_to_direction');
+
+  generator.point_to_direction = function () {
     let value = window.parseFloat(this.getTitleValue('DIRECTION')) || 0;
-    return `Turtle.pointTo(${value}, 'block_id_${this.id}');\n`;
+    return `Turtle.pointTo('${this.getTitleValue('VALUE')}', ${value},
+        'block_id_${this.id}');\n`;
```

Comment:
I _think_ having a newline in the middle of the interpolated string like this will result in the string itself having a newline in the middle, which will show up in `show code` and which we don't want.